### ISSUE TITLE
Running plugin-ci on PHP 8.1.22 shows warnings

### ISSUE THREAD

**Issue by jgg9026** [moodlehq/moodle-plugin-ci#235]
Hi,

I'm getting the following when trying to use the moodle-plugin-ci.phar downloaded from the latest released version v4.1.2

`Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jonathan.garcia/Downloads/moodle-plugin-ci-2.phar/vendor/symfony/console/Helper/HelperSet.php on line 104`

Found this while trying to run it on Linux (happens also in MacOs) for PHP 8.1.

This error also appears when running any command like "save points" or "grunt".

Any hep will very appreciated.

**Comment by ewallah** [comment]
Are you using the l[atest version (4) ](https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml#L55)?

`composer create-project -n --no-dev --prefer-dist moodlehq/moodle-plugin-ci ci `**^4**

**Comment by jgg9026** [comment]
Hi,

Using that binary file will show no errors. 

About that box version... it is causing errors when trying to build a new PHAR.
I did change it to the point the PHAR gets generated but then the following will appear:

```php
PHP Fatal error:  Uncaught UnexpectedValueException: DirectoryIterator::__construct(phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/../Resources/): Failed to open directory: operation failed in phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php:152
Stack trace:
#0 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php(152): DirectoryIterator->__construct('phar:///Users/j...')
#1 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php(55): Symfony\Component\Console\Command\DumpCompletionCommand->getSupportedShells()
#2 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/Command.php(137): Symfony\Component\Console\Command\DumpCompletionCommand->configure()
#3 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(1136): Symfony\Component\Console\Command\Command->__construct()
#4 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(1312): Symfony\Component\Console\Application->getDefaultCommands()
#5 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(536): Symfony\Component\Console\Application->init()
#6 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/bin/moodle-plugin-ci(72): Symfony\Component\Console\Application->add(Object(MoodlePluginCI\Command\AddConfigCommand))
#7 /Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar(15): require('phar:///Users/j...')
#8 {main}
  thrown in phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php on line 152

Fatal error: Uncaught UnexpectedValueException: DirectoryIterator::__construct(phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/../Resources/): Failed to open directory: operation failed in phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php:152
Stack trace:
#0 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php(152): DirectoryIterator->__construct('phar:///Users/j...')
#1 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php(55): Symfony\Component\Console\Command\DumpCompletionCommand->getSupportedShells()
#2 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/Command.php(137): Symfony\Component\Console\Command\DumpCompletionCommand->configure()
#3 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(1136): Symfony\Component\Console\Command\Command->__construct()
#4 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(1312): Symfony\Component\Console\Application->getDefaultCommands()
#5 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Application.php(536): Symfony\Component\Console\Application->init()
#6 phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/bin/moodle-plugin-ci(72): Symfony\Component\Console\Application->add(Object(MoodlePluginCI\Command\AddConfigCommand))
#7 /Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar(15): require('phar:///Users/j...')
#8 {main}
  thrown in phar:///Users/jonathan.garcia/moodle-ci/moodle-plugin-ci/build/moodle-plugin-ci.phar/vendor/symfony/console/Command/DumpCompletionCommand.php on line 152                           
``` 

I thin I can try to fix it but If you have any advice on how to do it, that will be great. 

Thanks!

**Comment by stronk7** [comment]
Hi,

thanks for testing!

As said, I've already here a development branch that moves to current box version... in fact, I'm attaching now a `moodle-plugin-ci.phar.gz` version built using it. My preliminary tests here (with PHP 8.2) are ok, but still have to verify that it works for PHP 7.4 and other details.

I think that you can use it in the mean time and, if you find something wrong, please share here.

Once I've the new phar tested, I will create a pull request to incorporate the changes and a new release (4.1.3) as soon as possible.

Ciao :-)

[moodle-plugin-ci.phar.gz](https://github.com/moodlehq/moodle-plugin-ci/files/12550249/moodle-plugin-ci.phar.gz)

**Comment by jgg9026** [comment]
Awesome, thanks!

I'll let you know if I find something.

Again, thanks for your help.

**Comment by stronk7** [comment]
Hi,

with #236 already merged, we are planning a new release (4.1.3) to happen soon. That one will come with the updated PHAR.

BTW, there are a few things within the PHAR that continue not working (pre-existing, they didn't work in previous versions, like 3.x). When I've some time I will try to fix some of them.

Ciao :-)
